component/topology/client.go

```diff
@@ -0,0 +1,89 @@
+package topology
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb-dashboard/util/client/httpclient"
+	"github.com/pingcap/tidb-dashboard/util/client/pdclient"
+	"github.com/zhongzc/ng_monitoring/config"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+)
+
+type Client struct {
+	pdCli   *pdclient.APIClient
+	etcdCli *clientv3.Client
+	pdCfg   config.PD
+}
+
+func NewClient(cfg *config.Config) (*Client, error) {
+	pdCli, etcdCli, err := createClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{
+		pdCli:   pdCli,
+		etcdCli: etcdCli,
+		pdCfg:   cfg.PD,
+	}, nil
+}
+
+func (c *Client) reCreateClient(cfg *config.Config) {
+	pdCli, etcdCli, err := createClient(cfg)
+	if err != nil {
+		log.Error("recreate pd/etcd client failed", zap.Error(err))
+		return
+	}
+	err = c.etcdCli.Close()
+	if err != nil {
+		log.Error("close etcd client failed", zap.Error(err))
+	}
+	c.pdCfg = cfg.PD
+	c.pdCli = pdCli
+	c.etcdCli = etcdCli
+	log.Info("recreate pd/etcd client success")
+}
+
+func (c *Client) Close() error {
+	return c.etcdCli.Close()
+}
+
+func createClient(cfg *config.Config) (*pdclient.APIClient, *clientv3.Client, error) {
```

Comment:
Why not move this to TiDB Dashboard? This kind of logic seems to be something common.

Reply:
em... It is not common now...